### PR TITLE
Kernel: Implement TCP window scaling

### PR DESCRIPTION
**Pull request by gunnarbeutner**
**Kernel: Also set TCP MSS option for SYN|ACK packets**

This sets the MSS option when we accept client connections. Previously we'd only set this option when we initiate the connection.

**Kernel: Move code for handling lost packets into its own function**

**WIP: TCP Window Scale**

This adds support for sending the TCP window scale option. The window size calculation is all wrong atm.

### CODE REVIEW

File: Kernel/Net/TCP.h [hunk 1]
```diff
@@ -36,7 +36,34 @@ class [[gnu::packed]] TCPOptionMSS {
     NetworkOrdered<u16> m_value;
 };
 
-static_assert(sizeof(TCPOptionMSS) == 4);
```
Comment (majestrate) [inline Kernel/Net/TCP.h:39]: was the remove of this `static_assert` a possible mistake or am I missing something?

Comment (gunnarbeutner) [reply Kernel/Net/TCP.h:39]: Thanks, fixed. :)

File: Kernel/Net/TCPSocket.cpp [hunk 2]
```diff
@@ -183,25 +183,33 @@ KResult TCPSocket::send_ack(bool allow_duplicate)
...
+
     if (flags & TCPFlags::ACK) {
         m_last_ack_number_sent = m_ack_number;
-        m_last_ack_sent_time = kgettimeofday();
+        m_last_ack_sent_time = now;
         tcp_packet.set_ack_number(m_ack_number);
     }
 
+    if (m_window_size <= NumericLimits<u16>::max() - window_increase_step
+        && m_last_window_increase < kgettimeofday() - window_increase_interval) {
```
Comment (majestrate) [inline Kernel/Net/TCPSocket.cpp:208]: this conditional is looking really chonky, would it be possible to put it on a diet?

Comment (majestrate) [reply Kernel/Net/TCPSocket.cpp:208]: possibly replace `kgettimeofday` with `now`?

**Pull request closed without merging**